src/compiler/types.ts

```diff
@@ -3187,13 +3187,14 @@ namespace ts {
     export type AnyImportSyntax = ImportDeclaration | ImportEqualsDeclaration;
 
     /* @internal */
-    export type AnyImportOrReExport = AnyImportSyntax | ExportDeclaration;
+    export type AnyImportOrReExport = AnyImportSyntax | ExportDeclaration | ImportTypeNode;

```

Comment:
IIRC, this is used as a shorthand in the definition of some other aliases which should _not_ handle import type nodes (due to only handling statements) - those probably should be changed to not use it, since the type guards used with those aliases don't check for nonstatements.

Reply:
Yeah, `isAnyImportOrReExport`'s return type isn't accurate with this change. (The other uses within the core layer aren't so bad because they're used in a union alongside `ImportTypeNode` anyway)